In [1]:
# importowanie stats z modułu scipy - korzystasz z funkcji statystycznych dostępnych w bibliotece SciPy
from scipy import stats

# Zadanie 3 - Test dla wartości oczekiwanej pojedynczej prób

Wytrzymałość na ciśnienie wewnętrzne jest ważną charakterystyką jakościową szkła butelek.


Pewna rozlewnia chce zamówić butelki, których średnia wytrzymałość przewyższa 1.20 N/mm2.


Pobrano próbę losową 20 butelek, które następnie umieszczono w maszynie hydrostatycznej,
zwiększając ciśnienie aż do zniszczenia butelki i otrzymano następujące wyniki (w N/mm2
):

1.36, 1.14, 1.27, 1.15, 1.20, 1.29, 1.27, 1.18, 1.23, 1.36, 1.38, 1.37, 1.30, 1.21, 1.33, 1.28, 1.32,
1.29, 1.33, 1.25.


Zbadaj, czy na poziomie istotności równym 0,05 można przyjąć założenie o jego normalności
rozkładu?


## 1. Sformułowanie hipotez


H0: Rozklad jest normalny


H1: Rozklad nie jest normalny


## 2. Wybór testu statystycznego
- Test nieparametryczny
- Pojedyncza próba
- Skala ilorazowa

->  Test Shapiro-wilka

## 3. Poziom istotności  
α=0,05


## 4. Obliczenie statystyki testowej i p-value oraz podjęcie decyzji weryfikacyjnej

In [3]:
# Wagi netto próbki
butelki = [1.36, 1.14, 1.27, 1.15, 1.20, 1.29, 1.27, 1.18, 1.23, 1.36, 1.38, 1.37, 1.30, 1.21, 1.33, 1.28, 1.32,
1.29, 1.33, 1.25]

#Test shapiro-wilka
statystyka, p_wartosc = stats.shapiro(butelki)
print("Wartość p: ", p_wartosc)
if p_wartosc > 0.05:
    print("Nie ma podstaw do odrzucenia hipotezy zerowej - dane pochodzą z rozkładu normalnego.")
else:
    print("Odrzucamy hipotezę zerową - dane nie pochodzą z rozkładu normalnego.")


Wartość p:  0.44210907987875553
Nie ma podstaw do odrzucenia hipotezy zerowej - dane pochodzą z rozkładu normalnego.


Z otrzymanego wydruku możemy odczytać p-wartość = 0.44210907987875553. \
Ponieważ: p>α, odrzucamy hipoteze zerowa. Nie ma podstaw do odrzucenia hipotezy zerowej - dane pochodzą z rozkładu normalnego.


## 5. Wnioski:
Dane pochodzą z rozkładu normalnego.


